woocommerce/woocommerce
# Update generic incompatibility notice

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #42072.

In #42724 PR, we've moved the switch to classic shortcode block button to separate components 

In this PR, we're updating the generic incompatibility notice message and adding the switch to classic shortcode block button.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Deactiavte any incompatible plugin like PayTm
2. Edit the Cart block page.
3. Focus on the Cart block. See the generic incompatibility notice in the sidebar. Click the "switch to classic cart" button.
4. Modal should open. Click switch. Block should be replaced.
5. In the snackbar bottom left, click "undo". Should revert back to blocks.
6. Repeat the steps for the Checkout Block page editor.

| Before | After |
|---|---|
|<img width="100%" alt="image" src="https://github.com/woocommerce/woocommerce/assets/11503784/c332bd1c-dc71-466f-b756-52e1e08ec0f0">|<img width="100%" alt="image" src="https://github.com/woocommerce/woocommerce/assets/11503784/7a21fa40-e048-4b8e-8488-d24523bd8e04">|

### For developer testing only 
1. Deactiavte any incompatible plugin like PayTm
2. Edit the Cart block page.
3. In browser console run `localStorage.debug = 'wc-admin:tracks';`
4. This will enable the logs for events in browser console. 
5. Focus on the Cart block. See the generic incompatibility notice in the sidebar. Click the "switch to classic cart" button.
4. Modal should open. Click switch. Block should be replaced.
5. Click on the undo button. 
6. In the browser console, confirm the following events are printed with `type: generic_notice`:
- wcadmin_switch_to_classic_shortcode_click
- wcadmin_switch_to_classic_shortcode_confirm
- wcadmin_switch_to_classic_shortcode_undo

![image](https://github.com/woocommerce/woocommerce/assets/11503784/637aaf5a-267e-41b3-a5cd-c4beac93caf8)


7. Enable PayTm or any incompatible plugin, confirm the notice is changed.
8. Focus on the Cart block. See the extension incompatibility notice in the sidebar. Click the "switch to classic cart" button.
9. Modal should open. Click switch. Block should be replaced.
10. Click on the undo button. 
11. In the browser console, confirm the following events are printed with `type: incompatible_notice`:
- wcadmin_switch_to_classic_shortcode_click
- wcadmin_switch_to_classic_shortcode_confirm
- wcadmin_switch_to_classic_shortcode_undo
<img width="1586" alt="image" src="https://github.com/woocommerce/woocommerce/assets/11503784/11eeae92-e5c2-447d-a04e-30cfae514428">



### Changelog entry




<details>

#### Significance




#### Type




#### Message 
Update the generic incompatibility notice message for the Cart and Checkout page.
 
#### Comment 

</details>
